airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -539,7 +539,7 @@ def expand_role(self, role: str) -> str:
         if "/" in role:
             return role
         else:
-            session, endpoint_url = self._get_credentials()
+            session, endpoint_url = self._get_credentials(region_name=None)
```

Comment:
Do we want to add `region_name` to this?

Reply:
I don't think so...
Also lets not add more joy to this party :) 
adding region_name is enhancement rather than addressing the bug/warnings problem so if someone thinks it's needed we can have it in a separated PR

Reply:
Since `_get_credentials` itself described region_name via hint as `Optional[Str]`... Should we instead add default = None for it in arguments?

Reply:
Doesn’t really matter since it’s internal; I like being explicit, so let’s not. (Actually I would’ve made it a keyword-only argument normally, but this is already a hot fix and I don’t want to complicate things unnecessarily.)